crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -1736,6 +1741,92 @@ mod tests {
         assert_eq!(rawev_id(room.latest_event().unwrap().event().clone()), "$a");
     }
 
+    #[async_test]
+    async fn test_recency_timestamp_is_found_when_processing_sliding_sync_response() {
+        // Given a logged-in client
+        let client = logged_in_base_client(None).await;
+        let room_id = room_id!("!r:e.uk");
+
+        // When I send sliding sync response containing a room with a recency timestamp
+        let room = {
+            let mut room = v4::SlidingSyncRoom::new();
+            room.timestamp = Some(MilliSecondsSinceUnixEpoch(42u32.into()));
+
+            room
+        };
```

Comment:
nit: here and below a few times, we could use `assign!()` to avoid these 6 lines :eyes:

Reply:
Done.